### DISCUSSION TITLE
Allow configuring s3 `etcd-snapshot-retention` in config file

### DISCUSSION THREAD

**Discussion by maggie44** [mysticaltech/terraform-hcloud-kube-hetzner#1360] [Ideas]
### Description

k3s has a number of options for configuring s3: https://docs.k3s.io/cli/etcd-snapshot#options

These can be configured by adding the flags, but ideally they would be added to the `kube.tf` config file as specific options. 

This has become particularly pertinent as a PR introduced back in August (https://github.com/k3s-io/k3s/issues/9866) changed the way s3 backups are done and for a standard 3 control plane cluster the default k3s value of `5` is likely insufficient for most as it no longer backs up all of the control planes for a sufficient time span. Instead, we may need to start scaling that number based on the number of control planes we run.

**Comment by mysticaltech** [comment]
@maggie44 PR please.

**Comment by maggie44** [comment]
Not forgotten, just not got to yet. 👍

**Comment by maggie44** [comment]
@mysticaltech, would:

```
 control_planes_custom_config = {
    "etcd-snapshot-retention" : 10,
  }
```

achieve this? The equivalent of passing in `--etcd-snapshot-retention value`? (https://docs.k3s.io/cli/etcd-snapshot#options)
